HLTrigger/JetMET/src/HLTDiJetEtaTopologyFilter.cc

```diff
@@ -0,0 +1,165 @@
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include<typeinfo>
+
+//
+// constructors and destructor
+//
+template<typename T>
+HLTDiJetEtaTopologyFilter<T>::HLTDiJetEtaTopologyFilter(const edm::ParameterSet& iConfig) : HLTFilter(iConfig),
+    inputJetTag_ (iConfig.template getParameter< edm::InputTag > ("inputJetTag")),
+    minPtAve_    (iConfig.template getParameter<double> ("minPtAve")),
+    minDphi_     (iConfig.template getParameter<double> ("minDphi")),
+    tagEtaMin_     (iConfig.template getParameter<double> ("minTagEta")),
+    tagEtaMax_     (iConfig.template getParameter<double> ("maxTagEta")),
+    probeEtaMin_     (iConfig.template getParameter<double> ("minProbeEta")),
+    probeEtaMax_     (iConfig.template getParameter<double> ("maxProbeEta")),
+    applyAbsToTag_ (iConfig.template getParameter<bool> ("applyAbsToTag")),
+    applyAbsToProbe_ (iConfig.template getParameter<bool> ("applyAbsToProbe")),
+    oppositeEta_ (iConfig.template getParameter<bool> ("oppositeEta")),
+
+    triggerType_ (iConfig.template getParameter<int> ("triggerType"))
+{
+    m_theJetToken = consumes<std::vector<T>>(inputJetTag_);
+    LogDebug("") << "HLTDiJetEtaTopologyFilter: Input/minDphi/triggerType : "
+        << inputJetTag_.encode() << " "
+        //<< minPtJet3_ << " "
+        << minDphi_ << " "
+        << triggerType_;
+}
+
+template<typename T>
+HLTDiJetEtaTopologyFilter<T>::~HLTDiJetEtaTopologyFilter(){}
+
+template<typename T>
+void
+HLTDiJetEtaTopologyFilter<T>::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    makeHLTFilterDescription(desc);
+    desc.add<edm::InputTag>("inputJetTag",edm::InputTag("hltIterativeCone5CaloJets"));
+    desc.add<double>("minPtAve",0.0);
+    desc.add<double>("atLeastOneJetAbovePT",0.0)->setComment("At least one jet with pt above threshold");
+    desc.add<double>("minPtTag",50.0)->setComment("pt requirement on tag jet");
+    desc.add<double>("minPtProbe",50.0)->setComment("pt requirement on probe jet");
+    //desc.add<double>("minPtJet3",99999.0);
+    desc.add<double>("minDphi",-1.0);
+    desc.add<double>("minTagEta", -1.);
+    desc.add<double>("maxTagEta", 1.4);
+    desc.add<double>("minProbeEta", 2.7);
+    desc.add<double>("maxProbeEta", 5.5);
+    desc.add<bool> ("applyAbsToTag", false),
+    desc.add<bool> ("applyAbsToProbe", false),
+    desc.add<bool> ("oppositeEta", false),
+    desc.add<int>("triggerType",trigger::TriggerJet);
+    descriptions.add(std::string("hlt")+std::string(typeid(HLTDiJetEtaTopologyFilter<T>).name()),desc);
```

Comment:
The synatx here has changed for 74X, it must be like
```
descriptions.add(defaultModuleLabel<HLTAlphaTFilter<T>>(), desc);
```
of course with the ciorrect class name.